tests/integration_test.rs

```diff
@@ -122,3 +189,25 @@ fn assert_ok_response(response: Result<ureq::Response, ureq::Error>) -> ureq::Re
 
     response
 }
+
+fn assert_response_status_code(response: Result<ureq::Response, ureq::Error>, expected_code: u16) {
+    match response {
+        #[allow(clippy::manual_assert)]
+        Err(ureq::Error::Status(code, response)) => {
+            if code != expected_code {
+                panic!(
+                    "Request failed with unexpected status code. Wanted: {expected_code} Found: {code}. Body:\n{}\n==EOF==",
+                    response.into_string().expect("cannot read response body"),
+                );
```

Comment:
I don't want to rewrite this `if` + `panic!` into `assert!` suggested by clippy, because the code building the assertion error message has side-effects - it reads the response body.

Reply:
Actually, quoting from `assert!()` docs:

```rs
    /// This macro has a second form, where a custom panic message can
    /// be provided with or without arguments for formatting. See [`std::fmt`]
    /// for syntax for this form. Expressions used as format arguments will only
    /// be evaluated if the assertion fails.
```